x-pack/legacy/plugins/ml/public/components/chart_tooltip/chart_tooltip_service.js

```diff
@@ -4,40 +4,40 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { BehaviorSubject } from 'rxjs';
 
+const doc = document.documentElement;
 
-import $ from 'jquery';
+const chartTooltipDefaultState = {
+  isTooltipVisible: false,
+  tooltipData: [],
+  tooltipPosition: { transform: 'translate(0px, 0px)' }
+};
 
-const doc = document.documentElement;
-const FADE_TIMEOUT_MS = 200;
+export const chartTooltip$ = new BehaviorSubject(chartTooltipDefaultState);
 
 export const mlChartTooltipService = {
   element: null,
-  fadeTimeout: null,
-  visible: false
 };
 
-mlChartTooltipService.show = function (contents, target, offset = { x: 0, y: 0 }) {
+mlChartTooltipService.show = function (tooltipData, target, offset = { x: 0, y: 0 }) {
   if (this.element === null || typeof target === 'undefined') {
     return;
   }
 
-  this.visible = true;
-  // if a previous fade out was happening, stop it
-  if (this.fadeTimeout !== null) {
-    clearTimeout(this.fadeTimeout);
-  }
+  // side bar width
+  const euiNavDrawer = document.getElementsByClassName('euiNavDrawer');
```

Comment:
is it possible to use `ref` here instead of imperatively access the DOM?

Reply:
That element is in a completely different area in Kibana we don't control. I'll leave this as-is replacement for the previous jQuery-based version in here for now.